src/coprocessor/dag/expr/builtin_cast.rs

```diff
@@ -94,15 +94,41 @@ impl FnCall {
     }
 
     pub fn cast_time_as_int(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<i64>> {
-        unimplemented!()
+        let val = try!(self.children[0].eval_time(ctx, row));
+        if val.is_none() {
+            return Ok(None);
+        }
+        let mut val = val.unwrap();
+        try!(val.round_frac(mysql::DEFAULT_FSP));
+        let dec = try!(val.to_decimal());
+        let res = try!(convert::dec_to_i64(dec));
```

Comment:
@XuHuaiyu it seems `tidb` doesn't consider `ctx.ignore_truncate` or `ctx.truncate_as_warning` ?